homeassistant/components/watergate/sensor.py

```diff
@@ -0,0 +1,234 @@
+        else None,
+        translation_key="water_temperature",
+        key="water_temperature",
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.telemetry.pressure if data.telemetry else None,
+        translation_key="water_pressure",
+        key="water_pressure",
+        native_unit_of_measurement=UnitOfPressure.MBAR,
+        device_class=SensorDeviceClass.PRESSURE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: (data.telemetry.flow / 1000)
+        if (data.telemetry and data.telemetry.flow is not None)
+        else None,
+        translation_key="water_flow_rate",
+        key="water_flow_rate",
+        native_unit_of_measurement=UnitOfVolumeFlowRate.LITERS_PER_MINUTE,
+        device_class=SensorDeviceClass.VOLUME_FLOW_RATE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: dt_util.as_utc(
+            dt_util.now() - timedelta(seconds=data.state.uptime)
+        )
+        if data.state
+        else None,
+        translation_key="uptime",
+        key="uptime",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        device_class=SensorDeviceClass.TIMESTAMP,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: PowerSupplyMode(data.state.power_supply)
+        if data.state
+        else None,
+        translation_key="power_supply",
+        key="power_supply",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        device_class=SensorDeviceClass.ENUM,
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: WatergateConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up all entries for Watergate Platform."""
+
+    coordinator = config_entry.runtime_data
+
+    async_add_entities(
+        SonicSensor(coordinator, description) for description in DESCRIPTIONS
+    )
+
+
+class SonicSensor(WatergateEntity, SensorEntity):
+    """Define a Sonic Sensor entity."""
+
+    entity_description: WatergateSensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: WatergateDataCoordinator,
+        entity_description: WatergateSensorEntityDescription,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator, entity_description.key)
+        self.entity_description = entity_description
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return (
+            super().available
+            and self.entity_description.value_fn(self.coordinator.data) is not None
+        )
```

Comment:
Why can't data be `None`?

Reply:
If it is None it means that there was an issue with fetching those data.